### PR TITLE
Improve transaction BBE 

### PR DESCRIPTION
**Pull request by niveathika**
## Purpose
$subject

Fixes: https://github.com/ballerina-platform/ballerina-standard-library/issues/3475

### CODE REVIEW

File: examples/mysql-atomic-transaction/mysql_atomic_transaction.bal [hunk 1]
```diff
@@ -13,23 +13,14 @@ public function main() returns error? {
...
-        } else {
-            error? err = commit;
-            if err is error {
-                log:printError("Error occurred while committing", err);
-            }
-        }
+        check commit;
+    } on fail error e {
+        log:printError(e.message());
+        log:printInfo("One of the queries failed. Rollback transaction.");
```
Comment (daneshk) [inline examples/mysql-atomic-transaction/mysql_atomic_transaction.bal:23]: Don't we need to rollback here

File: examples/jdbc-atomic-transaction/jdbc_atomic_transaction.bal [hunk 2]
```diff
@@ -14,23 +13,14 @@ public function main() returns error? {
...
-            }
-        }
+
+        _ = check jdbcClient->execute(`INSERT INTO Customers (firstName, lastName, registrationID, creditLimit, country)
+                                        VALUES ('Peter', 'Stuart', 4, 5000.75, 'USA')`);
+
+        check commit;
+    } on fail error e {
+        log:printError(e.message());
+        log:printInfo("One of the queries failed. Rollback transaction.");
```
Comment (daneshk) [inline examples/jdbc-atomic-transaction/jdbc_atomic_transaction.bal:23]: Don't we need to rollback here

File: examples/mysql-atomic-transaction/mysql_atomic_transaction.bal (hunk 1 above)
Comment (niveathika) [reply examples/mysql-atomic-transaction/mysql_atomic_transaction.bal:23]: Rollback is called automatically if the statements return error from check statements

File: examples/jdbc-atomic-transaction/jdbc_atomic_transaction.bal (hunk 2 above)
Comment (niveathika) [reply examples/jdbc-atomic-transaction/jdbc_atomic_transaction.bal:23]: Rollback is called automatically if the statements return error from check statements

**Review by daneshk — APPROVED** [review]

**Pull request merged**